G1ANT.Addon.Net/SimplifiedMessageSummary.cs

```diff
@@ -1,27 +1,321 @@
 ﻿using MailKit;
 using MimeKit;
-using G1ANT.Language;
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections.Generic;
+using G1ANT.Addon.Net.Models;
 
 namespace G1ANT.Addon.Net
 {
-    public class SimplifiedMessageSummary : MessageSummary
+    public class SimplifiedMessageSummary
     {
-        public new string TextBody;
         public IMailFolder Folder;
-        public ListStructure MessageAttachments;
 
-        public SimplifiedMessageSummary(MessageSummary summary, IMailFolder folder, ListStructure attachmentsList) : base(summary.Index)
+        private MessageSummary messageSummary = null;
+        private MimeMessage fullMessage = null;
+
+        public SimplifiedMessageSummary()
         {
-            TextBody = ParseBody(summary, folder);
-            MessageAttachments = attachmentsList;
-            Envelope = summary.Envelope;
-            UniqueId = summary.UniqueId;
+            FullMessage = new MimeMessage();
+        }
+
+        public SimplifiedMessageSummary(MessageSummary summary, IMailFolder folder)
+        {
+            messageSummary = summary;
             Folder = folder;
         }
 
-        private static string ParseBody(IMessageSummary message, IMailFolder folder)
+        protected SimplifiedMessageSummary(MimeMessage message)
+        {
+            FullMessage = message;
+        }
+
+        public MimeMessage FullMessage
+        {
+            get
+            {
+                if (fullMessage == null && messageSummary != null)
+                    fullMessage = Folder.GetMessage(messageSummary.UniqueId);
+                return fullMessage;
+            }
+            set
+            {
+                fullMessage = value;
+            }
+        }
+
+        public List<object> Attachments
+        {
+            get => CreateAttachmentStructuresFromAttachments(FullMessage.Attachments);
+            set
+            {
+                var bodyBuilder = new BodyBuilder();
+                if (bodyBuilder.Attachments != null)
+                {
+                    bodyBuilder.HtmlBody = HtmlBody;
+                    bodyBuilder.TextBody = TextBody;
```

Comment:
If there are no attachments unrelated properties (HtmlBody, TextBody, Body) won't be set. Looks weird for me.
Another weird thing is that you create `BodyBuilder` (line 53) without any parameters and one line below you expect that it will contain attachments. How is that possible? I'm not going to investigate how `BodyBuilder` works, I just want to show you that either lines 53-54 contain a bug or this method works in a weird way.

Reply:
If Attachment property of BodyBuilder is null in some reason the attachments will not be included into the FullMessage.Body which can contain Text, HtmlText and Attachments of the mail message. It looks fine 4 me